internal/controller/promotionstrategy_controller.go

```diff
@@ -303,62 +290,44 @@ func (r *PromotionStrategyReconciler) copyCommitStatuses(ctx context.Context, cs
-					status.Labels["promoter.argoproj.io/commit-status-copy-from-sha"] = utils.KubeSafeLabel(ctx, copyFromActiveHydratedSha)
-					status.Labels["promoter.argoproj.io/commit-status-copy-from-branch"] = utils.KubeSafeLabel(ctx, branch)
-					err := r.Create(ctx, status)
-					if err != nil {
-						return err
-					}
-				} else {
-					logger.Error(errGet, "failed to get CommitStatus", "namespace", proposedCSObjectKey.Namespace, "name", proposedCSObjectKey.Name)
-					return errGet
-				}
-			}
-			commitStatus.Spec.DeepCopyInto(&cs.Spec)
 
-			if cs.Labels == nil {
-				cs.Labels = make(map[string]string)
+			copiedCommitStatus = &promoterv1alpha1.CommitStatus{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:        proposedCSObjectKey.Name,
+					Annotations: commitStatus.Annotations,
+					Labels:      commitStatus.Labels,
+					Namespace:   commitStatus.Namespace,
+				},
+				Spec: promoterv1alpha1.CommitStatusSpec{
+					RepositoryReference: commitStatus.Spec.RepositoryReference,
+					Sha:                 copyToProposedHydratedSha,
+					Name:                branch + " - " + commitStatus.Spec.Name,
+					Description:         commitStatus.Spec.Description,
+					Phase:               commitStatus.Spec.Phase,
+					Url:                 "https://github.com/" + commitStatus.Spec.RepositoryReference.Owner + "/" + commitStatus.Spec.RepositoryReference.Name + "/commit/" + copyFromActiveHydratedSha,
+				},
+			}
+			if copiedCommitStatus.Labels == nil {
+				copiedCommitStatus.Labels = make(map[string]string)
 			}
-			cs.Labels[promoterv1alpha1.CommitStatusLabelCopy] = "true"
-			cs.Labels["promoter.argoproj.io/commit-status-copy-from"] = utils.KubeSafeLabel(ctx, commitStatus.Spec.Name)
-			cs.Labels["promoter.argoproj.io/commit-status-copy-from-sha"] = utils.KubeSafeLabel(ctx, copyFromActiveHydratedSha)
-			cs.Labels["promoter.argoproj.io/commit-status-copy-from-branch"] = utils.KubeSafeLabel(ctx, branch)
-			cs.Spec.Sha = copyToProposedHydratedSha
-			cs.Spec.Name = branch + " - " + commitStatus.Spec.Name
-
-			//TODO: This should be a function that gets the URL from the provider in a the providers required format
-			cs.Spec.Url = "https://github.com/" + cs.Spec.RepositoryReference.Owner + "/" + cs.Spec.RepositoryReference.Name + "/commit/" + copyFromActiveHydratedSha
-			err = r.Update(ctx, &cs)
+			copiedCommitStatus.Labels[promoterv1alpha1.CommitStatusLabelCopy] = "true"
+			copiedCommitStatus.Labels["promoter.argoproj.io/commit-status-copy-from"] = utils.KubeSafeLabel(ctx, commitStatus.Spec.Name)
+			copiedCommitStatus.Labels["promoter.argoproj.io/commit-status-copy-from-sha"] = utils.KubeSafeLabel(ctx, copyFromActiveHydratedSha)
+			copiedCommitStatus.Labels["promoter.argoproj.io/commit-status-copy-from-branch"] = utils.KubeSafeLabel(ctx, branch)
+
+			err = r.Patch(ctx, copiedCommitStatus, client.MergeFrom(&promoterv1alpha1.CommitStatus{}))
```

Comment:
Ditto from above, patch rather than Update?

Reply:
Same logic we own this so we can patch to avoid conflicts with commitstatus controller.